tensorboard/plugins/graph/tf_graph_controls/tf-graph-controls.ts

```diff
@@ -12,397 +12,546 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    },
+    /**
+     * @type {?Object<string, boolean>}
+     */
+    devicesForStats: {
+      value: null,
+      type: Object,
+      notify: true,
+      // TODO(stephanwlee): Change readonly -> readOnly and fix the setter.
+      readonly: true,
+    },
+    /**
+     * @type {!tf.graph.controls.ColorBy}
+     */
+    colorBy: {
+      type: String,
+      value: ColorBy.STRUCTURE,
+      notify: true,
+    },
+    colorByParams: {
+      type: Object,
+      notify: true,
+      // TODO(stephanwlee): Change readonly -> readOnly and fix the setter.
+      readonly: true,
+    },
+    datasets: {
+      type: Array,
+      observer: '_datasetsChanged',
+      value: () => [],
+    },
+    /**
+     * @type {tf.graph.render.RenderGraphInfo}
+     */
+    renderHierarchy: {
+      type: Object,
+      notify: true,
+    },
+    /**
+     * @type {!Selection}
+     */
+    selection: {
+      type: Object,
+      notify: true,
+      readOnly: true,
+      computed: '_computeSelection(datasets, _selectedRunIndex, _selectedTagIndex, _selectedGraphType)',
+    },
+    selectedFile: {
+      type: Object,
+      notify: true,
+    },
+    _selectedRunIndex: {
+      type: Number,
+      value: 0,
+      observer: '_selectedRunIndexChanged',
+    },
+    _selectedTagIndex: {
+      type: Number,
+      value: 0,
+      observer: '_selectedTagIndexChanged',
+    },
+    /**
+     * @type {tf.graph.SelectionType}
+     */
+    _selectedGraphType: {
+      type: String,
+      value: tf.graph.SelectionType.OP_GRAPH,
+    },
+    selectedNode: {
+      type: String,
+      notify: true,
+    },
+    _currentDevices: {
+      type: Array,
+      computed: '_getCurrentDevices(devicesForStats)',
+    },
+    _currentDeviceParams: {
+      type: Array,
+      computed: '_getCurrentDeviceParams(colorByParams)',
+    },
+    _currentXlaClusterParams: {
+      type: Array,
+      computed: '_getCurrentXlaClusterParams(colorByParams)',
+    },
+    _currentGradientParams: {
+      type: Object,
+      computed: '_getCurrentGradientParams(colorByParams, colorBy)',
+    },
+    showSessionRunsDropdown: {
+      type: Boolean,
+      value: true,
+    },
+    showUploadButton: {
+      type: Boolean,
+      value: true,
+    },
+    // This stores whether the feature for showing health pills is enabled in the first place.
+    healthPillsFeatureEnabled: Boolean,
+    // This stores whether to show health pills. Only relevant if healthPillsFeatureEnabled. The
+    // user can toggle this value.
+    healthPillsToggledOn: {
+      type: Boolean,
+      notify: true,
+    },
+    _legendOpened: {
+      type: Boolean,
+      value: true,
+    },
+  },
+
+  listeners: {
+    'trace-inputs.change': '_traceInputToggleChanged',
+  },
+
+  _traceInputToggleChanged: function(event: Event): void {
+    // Flip the state of the trace inputs flag.
+    // Needs to be casted to PaperToggleButton but constructor is missing.
+    const PaperToggleButton: any = event.target;
```

Comment:
Surely `PaperToggleButton` would be the constructor—it feels misleading
to give the _instance_ that name instead?

Could just do `(event as any).active`.

Reply:
removed the word `paper` but kept the code to make it clear what an expected `event.target` is.